src/zalsa.rs

```diff
@@ -77,7 +79,7 @@ pub struct IngredientIndex(u32);
 impl IngredientIndex {
     /// Create an ingredient index from a usize.
     pub(crate) fn from(v: usize) -> Self {
-        assert!(v < (u32::MAX as usize));
+        assert!(v <= u32::MAX as usize);
```

Comment:
I believe both of these checks (here and for `MemoIngredientIndex`) were accidentally wrong